src/libmongoc/src/mongoc/mongoc-topology-description.c

```diff
@@ -690,29 +692,28 @@ _calc_effective_read_mode (const mongoc_topology_description_t *td,
    switch (optype) {
    case MONGOC_SS_WRITE:
       /* We don't deal with write operations */
-      return MONGOC_READ_UNSET;
+      return false;
    case MONGOC_SS_READ:
       /* Maintain the requested read mode if it is a regular read operation */
-      return requested_read_mode;
+      return false;
    case MONGOC_SS_AGGREGATE_WITH_WRITE: {
       /* Check if any of the servers are too old to support the
        * aggregate-with-write on a secondary server */
       bool any_too_old = false;
       mongoc_set_for_each_const (mc_tpld_servers_const (td),
-                                 _check_any_server_less_than_wire_version_v5_0,
+                                 _check_any_server_less_than_wire_version_13,
                                  &any_too_old);
       if (any_too_old) {
          /* Force the read preference back to reading from a primary server, as
           * one or more servers in the system may not support the operation */
-         return MONGOC_READ_PRIMARY;
+         return true;
       }
       /* We're okay to send an aggr-with-write to a secondary server, so permit
        * the caller's read mode preference */
-      return requested_read_mode;
+      return false;
    }
    default:
-      BSON_UNREACHABLE (
-         "Invalid mongoc_ss_optype_t for _calc_effective_read_mode()");
+      BSON_UNREACHABLE ("Invalid mongoc_ss_optype_t for _force_read_primary()");
```

Comment:
```suggestion
      BSON_UNREACHABLE ("Invalid mongoc_ss_optype_t for _must_use_primary()");
```
Consistency with name of data member.